lib/node_modules/@stdlib/string/base/replace-before/docs/types/index.d.ts

```diff
@@ -0,0 +1,52 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="@stdlib/types"/>
+
+/**
+* Replace the substring before the first occurrence of a specified search string.
+*
+* @param str - input string
+* @param search - search string
+* @param replacement - replacement string
+* @returns substring
+*
+* @example
+* var out = replaceBefore( 'beep boop', ' ', 'foo' );
+* // returns 'foo boop'
+*
+* @example
+* var out = replaceBefore( 'beep boop', 'p', 'foo' );
+* // returns 'foop boop'
+*
+* @example
+* var out = replaceBefore( 'Hello World!', '', 'foo' );
+* // returns 'fooHello world!'
+*
+* @example
+* var out = replaceBefore( 'Hello World!', 'XYZ', 'foo' );
+* // returns 'fooHello World!'
```

Comment:
Once the implementation is modified, these examples should be updated, as the return values will no longer be correct.